### ISSUE TITLE
toThrow(constructor, message)

### ISSUE THREAD

**Issue by mattphillips** [jest-community/jest-extended#157]
Clone of https://github.com/facebook/jest/issues/3659

```js
expect(()=>{
  throw new Error('1234');
}).toThrow(Error, 'hello');
```

All that needs working out is the name as `toThrow` is built into core and so this shouldn't clash with it

**Comment by benjaminkay93** [comment]
I like the idea of toThrowWithMessage

**Comment by mattphillips** [comment]
@natealcedo yup sounds good to me 👍

**Comment by natealcedo** [comment]
So I've actually started taking a look at how to implement this. I thought it would be a good idea to copy what jest's default `toThrow` matcher currently does.  On that note I have a couple of questions. 

1. Should we be doing something like this `toThrowWithError(error, message)` where `message` can either be a regex or a string as per what jest's `toThrow` does?
2. How strict do we want to be with argument validation. Should we raise an error when `message` has not been provided?

Just some of the thoughts I've been considering. I'll try to get in an initial pull request sometime next week so that you guys can verify what I've worked on and whether I'm going in the right direction.

**Comment by mattphillips** [comment]
@natealcedo I like the idea keeping this  close to Jest’s current API with regex support. 

As for the validation we should probably fail the matcher  with a nice error message, saying the argument is missing. 

At some point in the future I’m planning on adding validation throughout all of the matchers so don’t overthink it for now, perhaps just checking the message is supplied :smile:

**Comment by natealcedo** [comment]
Let’s close since #170 fixes this :)

**Comment by mattphillips** [comment]
Released as part of [v0.11.0](https://github.com/jest-community/jest-extended/releases/tag/v0.11.0) :tada:
